tests/main/views/test_buyers.py

```diff
@@ -953,6 +953,298 @@ def test_404_if_question_does_not_exist(self):
+        self.data_api_client.get_brief.return_value = self._setup_brief(user_id=234)
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-specialists/1234/review-source")
+        assert res.status_code == 404
+
+    def test_review_page_404s_if_brief_has_wrong_lot(self):
+        self.data_api_client.get_brief.return_value = self._setup_brief()
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-outcomes/1234/review")
+        assert res.status_code == 404
+
+    def test_review_source_page_404s_if_brief_has_wrong_lot(self):
+        self.data_api_client.get_brief.return_value = self._setup_brief()
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-outcomes/1234/review-source")
+        assert res.status_code == 404
+
+    def test_review_page_400s_if_unanswered_questions(self):
+        brief_json = self._setup_brief()
+        brief_json['briefs'].pop('essentialRequirements')
+        self.data_api_client.get_brief.return_value = brief_json
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-specialists/1234/review")
+        assert res.status_code == 400
+
+    def test_review_source_page_400s_if_unanswered_questions(self):
+        brief_json = self._setup_brief()
+        brief_json['briefs'].pop('essentialRequirements')
+        self.data_api_client.get_brief.return_value = brief_json
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-specialists/1234/review-source")
+        assert res.status_code == 400
+
+    def test_review_page_renders_tabs_and_iframes(self):
+        self.data_api_client.get_brief.return_value = self._setup_brief()
+
+        res = self.client.get("/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/"
+                              "digital-specialists/1234/review")
+
+        assert res.status_code == 200
+        page_html = res.get_data(as_text=True)
+        document = html.fromstring(page_html)
+        assert "This is how suppliers will see your requirements when they are published." in page_html
+        assert len(document.xpath('//div[@class="govuk-tabs"]//a[contains(text(), "Desktop")]')) == 1
+
+        expected_src_link = "/buyers/frameworks/digital-outcomes-and-specialists-4/requirements/" \
+                            "digital-specialists/1234/review-source"
+        assert bool(document.xpath(
+            "//iframe[@src=$u][@title=$t][@class=$c]",
+            u=expected_src_link,
+            t="Preview of the page on desktop or tablet",
+            c="dm-desktop-iframe"
+        )) == 1
```

Comment:
Wait wuh `bool(...) == 1`?

Reply:
It's true!

(Good spot - that should be a `len`)